graylog2-server/src/main/java/org/graylog2/filters/ExtractorFilter.java

```diff
@@ -70,34 +83,43 @@ public boolean filter(Message msg) {
         return false;
     }
 
-    private List<Extractor> loadExtractors(final String inputId) {
+    @Subscribe
+    public void handleInputCreate(final InputCreated event) {
+        LOG.debug("Load extractors for input <{}>", event.id());
+        scheduler.schedule(() -> loadExtractors(event.id()), 0, TimeUnit.SECONDS);
+    }
+
+    @Subscribe
+    public void handleInputDelete(final InputDeleted event) {
+        LOG.debug("Removing input from extractors cache <{}>", event.id());
+        extractors.remove(event.id());
+    }
+
+    @Subscribe
+    public void handleInputUpdate(final InputUpdated event) {
```

Comment:
I don't have any warnings in my IDE and also didn't see any warnings during the build.

![image](https://cloud.githubusercontent.com/assets/461/16953752/870bda5c-4dce-11e6-8f18-af00da590592.png)